semgrep-core/src/spacegrep/src/lib/Comment.ml

```diff
@@ -0,0 +1,163 @@
+(*
+   Strip comments and the like as a preprocessing phase.
+*)
+
+open Printf
+
+type delimiters = { start : string; end_ : string }
+
+type style = {
+  name : string;
+  start : string option;
```

Comment:
Why not Start of string | Delimited of string * string ?
This would avoid those extra style functions as you could build directly the value.

Reply:
What about this comment? An ADT seems simpler no?

Reply:
I want to allow at most one kind of each comment (end-of-line and multine) because I think it makes the configuration in yaml more natural. We could still use variants internally, though. Let me see.

Reply:
ok, I switched to variants resulting in the following savings:
```
 3 files changed, 38 insertions(+), 77 deletions(-)
```